apps/docs-snippets/src/guide/transactions/transaction-request.test.ts

```diff
@@ -0,0 +1,180 @@
+import type { Account, Coin, Resource } from 'fuels';
+import {
+  CreateTransactionRequest,
+  ScriptTransactionRequest,
+  arrayify,
+  ZeroBytes32,
+  Address,
+  BaseAssetId,
+  bn,
+  Predicate,
+  Provider,
+  FUEL_NETWORK_URL,
+  WalletUnlocked,
+} from 'fuels';
+
+import {
+  DocSnippetProjectsEnum,
+  getDocsSnippetsForcProject,
+} from '../../../test/fixtures/forc-projects';
+
+/**
+ * @group node
+ */
+describe('Transaction Request', () => {
+  let provider: Provider;
+
+  const { abiContents: scriptAbi, binHexlified: scriptBytecode } = getDocsSnippetsForcProject(
+    DocSnippetProjectsEnum.SUM_SCRIPT
+  );
+
+  const { abiContents: predicateAbi, binHexlified: predicateBytecode } = getDocsSnippetsForcProject(
+    DocSnippetProjectsEnum.SIMPLE_PREDICATE
+  );
+
+  const address = Address.fromRandom();
+
+  const message = {
+    assetId: BaseAssetId,
+    sender: address,
+    recipient: address,
+    nonce: '0x',
+    amount: bn(0),
+    daHeight: bn(0),
+  };
+  const coin: Coin = {
+    id: '0x',
+    assetId: BaseAssetId,
+    amount: bn(0),
+    owner: address,
+    maturity: 1,
+    blockCreated: bn(0),
+    txCreatedIdx: bn(0),
+  };
+
+  beforeAll(async () => {
+    provider = await Provider.create(FUEL_NETWORK_URL);
+  });
+
+  it('creates a transaction request from ScriptTransactionRequest', () => {
+    const scriptMainFunctionArguments = [1];
+
+    // #region transaction-request-1
+    // #import { ScriptTransactionRequest };
+
+    // Instantiate the transaction request using a ScriptTransactionRequest
+    const transactionRequest = new ScriptTransactionRequest({ script: scriptBytecode });
+    // Set the script main function arguments (can also be passed in the class constructor)
+    transactionRequest.setData(scriptAbi, scriptMainFunctionArguments);
```

Comment:
Given the structure of our documentation.

The typegen section is directly before this example. Is it worth syncing up the example from the [typegen section](https://docs.fuel.network/docs/fuels-ts/abi-typegen/using-generated-types/#script), with this example, to use the same `Script` and then building on the knowledge from there.

Then we can back link to that section and it creates a better flow between sections?

Reply:
Hmm I agree we can probably improve flow between sections, but I don't think moving transactions into basics is the answer. There's quite a few elements in this section that I don't think everyone needs to know about.

IMO the better approach is to link to transactions from typegen?

Reply:
I wasn't thinking moving the transactions into the basics section.

More building upon the example from the previous section, however, potentially something worth re-visiting later, once the revamp has been completed.

```ts
import { ScriptAbi__factory } from './types'; 

const request = new ScriptTransactionRequest({
  script: ScriptAbi__factory.bin,
})

request.setData(ScriptAbi__factory.abi, [
  // Add the inputs here
])
```

Happy for you to resolve this as you see fit 😄

Reply:
Will see if any others have opinions on this 👍🏻 I am just conscious of greying the lines between sections. But I agree we want flow.

Reply:
I've made some improvements to the flow based on @Dhaiwat10 comment, let me know what you think - https://github.com/FuelLabs/fuels-ts/pull/1961/commits/7b20becfd6431fc1f8ff726d812551ec6b66c2ff

Reply:
Nice one - the links are a good touch.